File: 11-ContainerWithMostWater/11-ContainerWithMostWater.cpp
// Last updated: 2/11/2026, 11:08:47 AM
class Solution {
public:
    int maxArea(vector<int>& height) {
        int maxarea = 0;
        int left = 0;
        int right = height.size()-1;

        while (left <right){
            if (height[left] > height[right]){
                int area = height[right] * (right-left);
                right--;
                if (area > maxarea) maxarea = area;
            }
            else{
                int area = height[left] * (right-left);
                left++;
                if (area > maxarea) maxarea = area;
            }
        }

        return maxarea;
    }
};

File: 11-ContainerWithMostWater/11-ContainerWithMostWater.py
# Last updated: 2/11/2026, 11:08:50 AM
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        maxarea = 0

        if(len(height) == 0):
            return 0

        while (left < right):
            area = (right - left) * min (height[right],height[left])
            if (height[right] > height[left]):
                left+=1
            else:
                right-=1
            maxarea = max(area, maxarea)

        return maxarea